python/surf/devices/silabs/_Si5345.py

```diff
@@ -0,0 +1,2733 @@
+            name         = "FDEC",
+            description  = "1 a rising edge will cause the selected MultiSynth to decrement the output frequency by the Nx_FSTEPW parameter. See registers 0x03390x0358",
+            offset       = (0x001D << 2),
+            bitSize      = 1,
+            bitOffset    = 1,
+            # hidden       = True,
+            function     = pr.BaseCommand.toggle,
+        ))         
+
+        self.add(pr.RemoteVariable(
+            name        = 'PDN',
+            description = '1 to put the device into low power mode',
+            offset      = (0x001E << 2),
+            base        = pr.Bool,
+            bitSize     = 1,
+            bitOffset   = 0,
+            mode        = 'RW',
+        ))
+        
+        self.add(pr.RemoteVariable(
+            name        = 'HARD_RST',
+            description = '1 causes hard reset. The same as power up except that the serial port access is not held at reset.',
+            offset      = (0x001E << 2),
+            base        = pr.Bool,
+            bitSize     = 1,
+            bitOffset   = 1,
+            mode        = 'RW',
+        ))        
+    
+        self.add(pr.RemoteCommand(  
+            name         = "SYNC",
+            description  = "1 to reset all output R dividers to the same state.",
+            offset       = (0x001E << 2),
+            bitSize      = 1,
+            bitOffset    = 2,
+            # hidden       = True,
+            function     = pr.BaseCommand.toggle,
+        ))    
+    
+        self.add(pr.RemoteVariable(
+            name        = 'SPI_3WIRE',
+            description = '0 for 4-wire SPI, 1 for 3-wire SPI',
+            offset      = (0x002B << 2),
+            base        = pr.Bool,
+            bitSize     = 1,
+            bitOffset   = 3,
+            mode        = 'RW',
+        )) 
+        
+        self.add(pr.RemoteVariable(
+            name        = 'AUTO_NDIV_UPDATE',
+            description = 'Set by CBPro.',
+            offset      = (0x002B << 2),
+            base        = pr.Bool,
+            bitSize     = 1,
+            bitOffset   = 5,
+            mode        = 'RW',
+            hidden      = simpleDisply,
+        ))    
+
+        self.add(pr.RemoteVariable(
+            name        = 'LOS_EN',
+            description = '1 to enable LOS for a clock input',
+            offset      = (0x002C << 2),
+            bitSize     = 4,
+            bitOffset   = 0,
+            mode        = 'RW',
+        ))
+
+        self.add(pr.RemoteVariable(
+            name        = 'LOSXAXB_DIS',
+            description = '0: Enable LOS Detection (default)',
+            offset      = (0x002C << 2),
+            base        = pr.Bool,
+            bitSize     = 1,
+            bitOffset   = 4,
+            mode        = 'RW',
+        ))        
+
+        for i in range(4):
+            self.add(pr.RemoteVariable(
+                name        = ('LOS_VAL_TIME[%d]' % i), 
```

Comment:
This style of string formatting is discouraged in new python3 code.
See discussion here:
https://stackoverflow.com/questions/13451989/pythons-many-ways-of-string-formatting-are-the-older-ones-going-to-be-deprec

I find f-strings to be much easier to make and more readable:
`name = f'LOS_VAL_TIME[{i}]',`

Reply:
Updated in  e28d8f3